HACKING.md

```diff
@@ -233,7 +233,13 @@ the build frontend is [build](https://pypi.org/project/build/).
 
 # Publishing a Release
 
-Before you start, make sure you are at the main branch of your forked operator repo, and your main branch is up-to-date. This is to ensure the release automation script is the latest.
+Before you start, ensure that your environment variable GITHUB_TOKEN is set and that the token has sufficient permissions. The easiest way to set a token is to run `gh auth login` first, follow the steps to log in, then run `export GITHUB_TOKEN=$(gh auth token)`.
+
+Alternatively, you can also create a classic token. To do so, go to GitHub -> Settings -> Developer Settings -> Personal access tokens -> Tokens (classic), and click "Generate new token (classic)" (shortcut: click [this link](https://github.com/settings/tokens/new)), and under "Select scopes", "repo", select "repo:status" and "public_repo". Then, set the environment variable `GITHUB_TOKEN` with the newly generated token.
+
+> Note: A fine-grained token will not work because if the owner is your GitHub account, it can't create a PR on the canonical repo, and if the owner is Canonical, it can't push to your forked repo. 
```

Comment:
This is annoying. It would be nice to not have a GitHub token on my Canonical device that can access non-Canonical organisations. Admittedly, I'm logged in to GitHub in the browser, but that is at least restricted in what it will do without 2FA, and it's not sitting around in an environment variable.

Not for this PR, but in the future would we be able to avoid this by letting people have *two* fine-grained tokens? Each GitHub command must work with a single org at a time, right? If some people are happy with a single broad token then they could just provide the same token twice, but more security-minded people could then limit what the script can do.

If that sounds reasonable, I'm happy to open an issue for it. Or maybe there are other ideas about how to make improvements here?

Reply:
Actually, thinking about this more, isn't all the pushing done with the `git` CLI rather than via the GitHub API? Wouldn't that mean that a fine-grained token restricted to only canonical/operator would work?